tools/data-handler/src/containers/project/resource-collector.ts

```diff
@@ -0,0 +1,359 @@
+   */
+  public changed() {
+    this.collectLocalResources();
+  }
+
+  /**
+   * Removes a resource from Project.
+   * @param resource Resource to remove.
+   */
+  public remove(resource: Resource) {
+    const { type } = resourceNameParts(resource.name);
+    let arrayToModify: Resource[];
+    switch (type) {
+      case 'cardTypes':
+        arrayToModify = this.localCardTypes;
+        break;
+      case 'fieldTypes':
+        arrayToModify = this.localFieldTypes;
+        break;
+      case 'linkTypes':
+        arrayToModify = this.localLinkTypes;
+        break;
+      case 'reports':
+        arrayToModify = this.localReports;
+        break;
+      case 'templates':
+        arrayToModify = this.localTemplates;
+        break;
+      case 'workflows':
+        arrayToModify = this.localWorkflows;
+        break;
+      default: {
+        throw new Error(
+          `Resource type '${type}' not handled in 'removeResource'`,
+        );
+      }
+    }
+    const index = arrayToModify.indexOf(resource, 0);
+    if (index > -1) {
+      arrayToModify.splice(index, 1);
+    }
+  }
+
+  /**
+   * Returns resource's metadata.
+   * @param type Type of resource (e.g. 'templates').
+   * @param name Name of the resource, in long name format.
+   * @param from Defines where resources are collected from.
+   * @returns Resources metadata, or undefined if resource was not found.
+   * @note that caller need to convert this to specific type (e.g "as unknown as Workflow")
+   */
+  public async resource(
+    type: string,
+    name: string,
+    from: ResourcesFrom,
+  ): Promise<object | undefined> {
+    if (!name) {
+      return undefined;
+    }
+    if (!name.endsWith('.json')) {
+      name += '.json';
+    }
+    const found = (await this.resources(type, from)).find(
+      (item) => item.name === name && item.path,
+    );
+    if (!found || !found.path) {
+      return undefined;
+    }
+    const file = await readJsonFile(join(found.path, basename(found.name)));
+    return file;
+  }
+
+  /**
+   * Checks if resource of 'type' with 'name' exists.
+   * @param type Type of resource (e.g. 'templates').
+   * @param name Name of the resource, in long name format.
+   * @returns true, if resource exits, false otherwise.
+   */
+  public async resourceExists(type: string, name: string): Promise<boolean> {
+    if (!name) {
+      return false;
+    }
+    return (await this.resources(type)).some((item) => item.name === name);
+  }
+
+  /**
+   * Returns resources of 'type'. Returned resources are either local, or from modules or all of them.
+   * @param type Type of resource (e.g. 'templates').
+   * @param from Defines where resources are collected from.
+   * @returns Array of resources.
+   */
+  public async resources(
+    type: string,
+    from: ResourcesFrom = ResourcesFrom.all,
+  ): Promise<Resource[]> {
+    const moduleResources =
+      from !== ResourcesFrom.localOnly
+        ? await this.addResourcesFromModules(type)
```

Comment:
Could we also store modules in memory? They are also easy to keep in sync, since only the import command updates them

Reply:
Yes, we could. Only when import/remove is used the cached values would change. 
I'll include that change in this PR.

Reply:
**Done** - now importing module calls `Project` to update the imported resources; who in turn calls `ResourceCollector` to make the updates.